src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -745,7 +745,17 @@ ExtractInsertPartitionValue(Query *query, Var *partitionColumn)
 												partitionColumn->varattno);
 	if (targetEntry != NULL)
 	{
-		Assert(IsA(targetEntry->expr, Const));
+		if (IsA(targetEntry->expr, Param))
+		{
+			ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+							errmsg("cannot execute a prepared insert statement on "
```

Comment:
Actually, while rebasing, I noticed that the issue #645 is fixed now in the planner with new commits of @lithp.
https://github.com/citusdata/citus/blame/master/src/backend/distributed/planner/multi_router_planner.c#L308

And the current error message is:
ERROR:  values given for the partition column must be constants or constant expressions

I will keep the new regression tests but will remove the fix.